### PR TITLE
KIALI-490: Add tests for service ordering by error rate

### PR DESCRIPTION
**Pull request by israel-hdez**
This is:
- Extracting code into components or classes. This is to isolate code and make it testable.
- Adding tests for the extracted code.

### CODE REVIEW

File: src/pages/ServiceList/RateIntervalToolbarItem.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+export default class RateIntervalToolbarItem extends React.Component<RateIntervalToolbarItemProps> {
+  render() {
+    let rateIntervalSelected = MetricsOptionsBar.RateIntervals.find(el => {
+      return el[0] === this.props.rateIntervalSelected;
+    });
+
+    return (
+      <div className="form-group">
+        <label>Rate Interval:&nbsp;</label>
```
Comment (cfcosta) [inline src/pages/ServiceList/RateIntervalToolbarItem.tsx]: No `&nbsp;`, please. If necessary add the padding/margin on the label scss.

File: src/pages/ServiceList/ServiceErrorRate.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+type ServiceErrorRateProps = {
+  service: ServiceItem;
+};
+
+export default class ServiceErrorRate extends React.Component<ServiceErrorRateProps> {
+  render() {
+    return (
+      <>
+        <strong>Error rate: </strong>
+        {this.props.service.request_count > 0
```
Comment (cfcosta) [inline src/pages/ServiceList/ServiceErrorRate.tsx]: This (and the other clause down) should be methods, I think. They are complex enough that it becomes weird when rendering with the mixed indenting.

File: src/pages/ServiceList/ServiceErrorRate.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          ? (this.props.service.error_rate * 100).toFixed(2) + '%'
+          : '(No requests)'}&nbsp;
+        <Icon
+          type="pf"
+          name={
+            this.props.service.error_rate > ERROR_THRESHOLD
```
Comment (cfcosta) [inline src/pages/ServiceList/ServiceErrorRate.tsx]: Same here.

File: src/pages/ServiceList/ServiceListComponent.tsx [hunk 4]
```diff
@@ -265,16 +238,7 @@ class ServiceListComponent extends React.Component<ServiceListComponentProps, Se
...
-              {serviceItem.request_count > 0 ? (serviceItem.error_rate * 100).toFixed(2) + '%' : '(No requests)'}&nbsp;
-              <Icon
-                type="pf"
-                name={
-                  serviceItem.error_rate > ERROR_THRESHOLD
-                    ? 'error-circle-o'
-                    : serviceItem.error_rate > WARNING_THRESHOLD ? 'warning-triangle-o' : 'ok'
-                }
-              />
+              <ServiceErrorRate service={serviceItem} />
```
Comment (cfcosta) [inline src/pages/ServiceList/ServiceListComponent.tsx]: Aye!

**Comment by israel-hdez** [comment]
@cfcosta Rebased and fixed your comments!

**Comment by mtho11** [comment]
Please squash.

File: src/pages/ServiceList/RateIntervalToolbarItem.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import { DropdownButton, MenuItem } from 'patternfly-react';
+
+type RateIntervalToolbarItemProps = {
+  rateIntervalSelected: string;
+  onRateIntervalChanged?: (key: string) => void;
+};
+
+export default class RateIntervalToolbarItem extends React.Component<RateIntervalToolbarItemProps> {
+  render() {
+    let rateIntervalSelected = MetricsOptionsBar.RateIntervals.find(el => {
```
Comment (mtho11) [inline src/pages/ServiceList/RateIntervalToolbarItem.tsx]: looks like the `let` can be a`const`

File: src/pages/ServiceList/ServiceErrorRate.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      <>
+        <strong>Error rate: </strong>
+        {this.errorRateIndicator()}
+        <Icon style={{ paddingLeft: '0.5em' }} type="pf" name={this.errorRateIconName()} />
+      </>
+    );
+  }
+
+  private errorRateIndicator = () => {
+    let rateText = '(No requests)';
```
Comment (mtho11) [inline src/pages/ServiceList/ServiceErrorRate.tsx]: This could be shortened to a ternary operator

File: src/types/ServiceListComponent.ts [hunk 7]
```diff
@@ -26,3 +26,9 @@ export interface ServiceItem {
 }
 
 export const IstioLogo = require('../assets/img/istio-logo.svg');
+
+export type SortField = {
```
Comment (mtho11) [inline src/types/ServiceListComponent.ts]: As a rule of thumb if you are going to export it then it should probably be an `interface` instead of `type`

**Comment by israel-hdez** [comment]
@mtho11 Squashed and fixed your comments.

**Pull request merged**